dev_tools/ci/size-labeler.sh

```diff
@@ -0,0 +1,184 @@
+#!/usr/bin/env bash
+# Copyright 2025 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Updates size labels on pull requests based on the number of lines changed.
+# Usage:
+#   label-pr-size.sh THE_PR_NUMBER
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+[[ "${BASH_VERSINFO[0]}" -ge 4 ]] || { echo "ERROR: Bash version 4+ required." >&2; exit 1; }
+
+set -euo pipefail -o errtrace
+shopt -s inherit_errexit
+
+declare -a LABELS=(
+    "Size: XS"
+    "Size: S"
+    "Size: M"
+    "Size: L"
+    "Size: XL"
+)
+
+declare -A LIMITS=(
+    ["${LABELS[0]}"]=10
+    ["${LABELS[1]}"]=50
+    ["${LABELS[2]}"]=200
+    ["${LABELS[3]}"]=800
+    ["${LABELS[4]}"]="$((2 ** 63 - 1))"
+)
+
+declare -a IGNORED=(
+    "*_pb2.py"
+    "*_pb2.pyi"
+    "*_pb2_grpc.py"
+    ".*.lock"
+)
+
+function info() {
+    echo >&2 "INFO: ${*}"
+}
+
+function error() {
+    echo >&2 "ERROR: ${*}"
+}
+
+function jq_stdin() {
+    local infile
+    infile="$(mktemp)"
+    readonly infile
+
+    cat >"${infile}"
+    jq_file "$@" "${infile}"
+}
+
+function jq_file() {
+    # Regardless of the success, store the return code.
+    # Prepend each sttderr with command args and send back to stderr.
+    jq "${@}" 2> >(echo "$(date -Iseconds) stderr from jq ${*}: " 1>&2) &&
+        rc="${?}" ||
+        rc="${?}"
+    if [[ "${rc}" != "0" ]]; then
+        error "The jq program failed: ${*}"
+        error "Note the quotes above may be wrong. Here was the (possibly empty) input in ${*: -1}:"
+        cat "${@: -1}" # Assumes last argument is input file!!
+        exit 1
```

Comment:
No need for the `exit` here.  This function returns the `rc` exit code from the `jq` run, let us leave it up to function caller to handle that.  Because this script has the `-e` option set (line 24), a plain call to `jq_file` would exit immediately if it returned a non-zero.



```suggestion
```

Reply:
This did not happen in my local testing in a shell; that's why I added the exit statement.

Reply:
I have a suspicion about why this happens. Investigating now and will comment back.

Reply:
That function has code for returning a non-zero exit code.  Either that code should be in or the `exit` statement, but not both.  The former case is frankly better, because it allows `jq_file` callers to catch and handle errors, which is not possible if script terminates.

Reply:
I'm not sure what happened before. My suspicion was that the error return was masked because the function was called inside command substitution `$(...)`, but this script sets `-e` and `inherit_errexit`.

Let's take out the `exit`.